### PR TITLE
#46 Grafana dashboard

### LINKED ISSUE #46: Add Graphite server to consume NoSQLBench data
NoSQLBench supports sending metrics directly to a graphite server: http://docs.nosqlbench.io/#/docs/reference/advanced_metrics.md

We can launch a Graphite server in the beginning of the workflow (or during the Helm chart installation) to collect the data generated by NSQLB.

┆Issue is synchronized with this [Jira Task](https://datastax-oss.atlassian.net/browse/AD-95) by [Unito](https://www.unito.io)
┆Issue Number: AD-95

### PR DESCRIPTION
**Pull request by grighetto**
Report performance metrics from both NoSQLBench jobs to a standalone graphite/grafana/prometheus setup.

Resolves #46

### CODE REVIEW

File: helm/adelphi/config/grafana/nb4-new-dashboard.json [hunk 1]
```diff
@@ -0,0 +1,1405 @@
+{
```
Comment (grighetto) [inline helm/adelphi/config/grafana/nb4-new-dashboard.json:1]: This is the default dashboard that comes with NB. We'll soon replace it with something that makes more sense for Adelphi, like some kind of side-by-side comparison with sensitive performance metrics. New ticket coming up for this.

Comment (grighetto) [reply helm/adelphi/config/grafana/nb4-new-dashboard.json:1]: https://github.com/datastax/adelphi/issues/91

File: helm/adelphi/templates/prometheus-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    alerting:
+      alertmanagers:
+      - static_configs:
+        - targets: []
+        scheme: http
+        timeout: 10s
+
+    scrape_configs:
+      - job_name: prometheus
```
Comment (jdonenine) [inline helm/adelphi/templates/prometheus-configmap.yaml:23]: I don't really think it much matters, but do we need prometheus scraping itself in this scenario? Not sure it really adds any overhead or not, but I'd imagine our users would never look at the metrics produced by doing this.

**Review by jdonenine — COMMENTED** [review]
Looks good @grighetto everything ran as expected and could be seen in Grafana, this is a great add for usability 🎉 .

Just two minor things I noted.

We'll also want to update the docs with info about this feature as well - let's cover that as a separate issue though and go ahead and get this in.

File: helm/adelphi/templates/prometheus-configmap.yaml (hunk 2 above)
Comment (grighetto) [reply helm/adelphi/templates/prometheus-configmap.yaml:23]: agreed, I'll remove it, that's coming from the stock NoSQLBench config

File: helm/adelphi/config/grafana/nb4-new-dashboard.json [hunk 3]
```diff
@@ -0,0 +1,1405 @@
...
+      "6h",
+      "12h",
+      "24h",
+      "2d",
+      "7d",
+      "30d"
+    ]
+  },
+  "timezone": "",
+  "title": "NB4 Dashboard",
```
[the remark this answers is not in the document]
Comment (grighetto) [reply helm/adelphi/config/grafana/nb4-new-dashboard.json]: Renamed to "Adelphi NoSQLBench" for now, we can change it again if needed.

File: helm/adelphi/templates/grafana.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+apiVersion: v1
+kind: Pod
```
[the remark this answers is not in the document]
Comment (grighetto) [reply helm/adelphi/templates/grafana.yaml:2]: k8s service added for grafana

File: helm/adelphi/templates/prometheus-configmap.yaml (hunk 2 above)
Comment (grighetto) [reply helm/adelphi/templates/prometheus-configmap.yaml]: Done

**Comment by grighetto** [comment]
@jdonenine Thanks for the quick review, I addressed all your comments and tested on GKE.
If you don't have any other feedback, I'll go ahead and merge this.

**Review by jdonenine — APPROVED** [review]
Thanks for the changes. Good to go!

**Pull request merged**